src/screens/Main.tsx

```diff
@@ -0,0 +1,28 @@
+import { useQuery } from "@tanstack/react-query";
+import { Product, getProductsQuery } from "../api/products";
+import { errorToast } from "../utils";
+import { ProductCard } from "../components";
+
+export function Main() {
+  const queryParam = {
+    ...getProductsQuery(),
+    onError: errorToast,
+  };
+  const { data: products, error, isPending } = useQuery(queryParam);
+
+  if (isPending) return "Loading...";
```

Comment:
Use `isLoading` instead, I think `isPending` is from a later version of `react-query`.